marker_api/src/lib.rs

```diff
@@ -7,10 +7,21 @@
 #![allow(clippy::missing_panics_doc)] // Temporary allow for `todo!`s
 #![allow(clippy::new_without_default)] // Not very helpful as `new` is almost always cfged
 #![cfg_attr(not(feature = "driver-api"), allow(dead_code))]
+#![cfg_attr(
+    marker,
+    feature(register_tool),
+    register_tool(marker),
+    // Allowed because it is triggered on a bunch of private/pub(crate) methods.
+    // There isn't a way to inspect the item visibility in this lint's impl yet.
+    // Once rust-marker/marker#26 is done and lint impl ignores private functions
```

Comment:
Yeah, that issue is becoming a real blocker. I've added it to the v0.4.0 milestone and plan to work on it soon :)